开始运行所有签到测试...
TRUNCATE TABLE

=== 运行测试部分1：基本签到功能 ===
BEGIN
DELETE 4
DELETE 5
DELETE 3
INSERT 0 3
INSERT 0 1
INSERT 0 1
INSERT 0 1
INSERT 0 1
INSERT 0 1
DO
         test_case         
---------------------------
 测试用例1：新会员首次签到
(1 row)

                                                                         handle_check_in                                                                          
------------------------------------------------------------------------------------------------------------------------------------------------------------------
 {"isExtra": true, "message": "您暂无有效的团课卡，已记录为额外签到", "success": true, "checkInId": "e1a9ffc7-c137-4b8b-94db-777836090bbc", "isNewMember": false}
(1 row)

              test_case              
-------------------------------------
 测试用例2：老会员团课签到（课时卡）
(1 row)

                                                            handle_check_in                                                            
---------------------------------------------------------------------------------------------------------------------------------------
 {"isExtra": false, "message": "签到成功", "success": true, "checkInId": "aaad5078-eece-40ed-a840-77f51164a99a", "isNewMember": false}
(1 row)

             test_case             
-----------------------------------
 测试用例3：老会员团课签到（月卡）
(1 row)

                                                            handle_check_in                                                            
---------------------------------------------------------------------------------------------------------------------------------------
 {"isExtra": false, "message": "签到成功", "success": true, "checkInId": "ee902c39-2680-4d97-8931-72631d1b3695", "isNewMember": false}
(1 row)

COMMIT

=== 运行测试部分2：私教课签到和额外签到 ===
BEGIN
             test_case              
------------------------------------
 测试用例4：老会员私教课签到（1v1）
(1 row)

                                                            handle_check_in                                                             
----------------------------------------------------------------------------------------------------------------------------------------
 {"message": "insert or update on table \"check_ins\" violates foreign key constraint \"check_ins_trainer_id_fkey\"", "success": false}
(1 row)

             test_case              
------------------------------------
 测试用例5：老会员私教课签到（1v2）
(1 row)

psql:test_sql/test_check_in_part2.sql:30: ERROR:  function handle_check_in(unknown, unknown, unknown, unknown, unknown, timestamp without time zone, unknown, unknown, boolean) does not exist
LINE 1: SELECT handle_check_in(
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
psql:test_sql/test_check_in_part2.sql:33: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part2.sql:45: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part2.sql:48: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part2.sql:57: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part2.sql:60: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part2.sql:70: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part2.sql:73: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part2.sql:83: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part2.sql:94: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part2.sql:97: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part2.sql:107: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part2.sql:110: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part2.sql:121: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part2.sql:124: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part2.sql:135: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part2.sql:138: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part2.sql:146: ERROR:  current transaction is aborted, commands ignored until end of transaction block
ROLLBACK

=== 运行测试部分3：特殊情况和边界条件 ===
BEGIN
DELETE 1
DELETE 0
DELETE 1
INSERT 0 2
INSERT 0 1
INSERT 0 1
            test_case             
----------------------------------
 测试用例11：使用今天过期的卡签到
(1 row)

                                                            handle_check_in                                                            
---------------------------------------------------------------------------------------------------------------------------------------
 {"isExtra": false, "message": "签到成功", "success": true, "checkInId": "7782da04-1115-4f70-b7ea-70253fbb0ea1", "isNewMember": false}
(1 row)

                test_case                
-----------------------------------------
 测试用例12：使用只剩1次课时的私教卡签到
(1 row)

                                                            handle_check_in                                                             
----------------------------------------------------------------------------------------------------------------------------------------
 {"message": "insert or update on table \"check_ins\" violates foreign key constraint \"check_ins_trainer_id_fkey\"", "success": false}
(1 row)

                         test_case                          
------------------------------------------------------------
 测试用例13：尝试再次使用已用完课时的私教卡签到（应该失败）
(1 row)

psql:test_sql/test_check_in_part3.sql:87: ERROR:  function handle_check_in(unknown, unknown, unknown, unknown, unknown, timestamp without time zone, unknown, unknown) does not exist
LINE 1: SELECT handle_check_in(
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
psql:test_sql/test_check_in_part3.sql:90: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part3.sql:99: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part3.sql:102: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part3.sql:111: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part3.sql:114: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part3.sql:124: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part3.sql:127: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part3.sql:137: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part3.sql:140: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part3.sql:150: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part3.sql:153: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part3.sql:163: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part3.sql:166: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part3.sql:176: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part3.sql:179: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part3.sql:183: ERROR:  current transaction is aborted, commands ignored until end of transaction block
ROLLBACK

=== 运行测试部分4：已存在会员的额外签到和名字匹配问题 ===
BEGIN
DELETE 0
DELETE 0
DELETE 0
INSERT 0 2
                   test_case                    
------------------------------------------------
 测试用例19：测试已存在会员的额外签到（通过ID）
(1 row)

psql:test_sql/test_check_in_part4.sql:39: ERROR:  function handle_check_in(unknown, unknown, unknown, unknown, unknown, date, unknown, unknown, unknown, boolean) does not exist
LINE 1: SELECT handle_check_in(
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
psql:test_sql/test_check_in_part4.sql:42: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part4.sql:54: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part4.sql:57: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part4.sql:69: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part4.sql:72: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part4.sql:84: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part4.sql:87: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part4.sql:99: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part4.sql:102: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part4.sql:114: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part4.sql:117: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part4.sql:123: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part4.sql:126: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part4.sql:131: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part4.sql:134: ERROR:  current transaction is aborted, commands ignored until end of transaction block
psql:test_sql/test_check_in_part4.sql:138: ERROR:  current transaction is aborted, commands ignored until end of transaction block
ROLLBACK

=== 测试结果摘要 ===

成功的签到记录:
    name    | check_in_date | class_type |  time_slot  | is_extra | trainer_name | is_1v2 
------------+---------------+------------+-------------+----------+--------------+--------
 测试会员D  | 2025-02-25    | morning    | 09:00-10:30 | f        |              | f
 测试新会员 | 2025-03-02    | morning    | 09:00-10:30 | t        |              | f
 测试老会员 | 2025-03-02    | morning    | 09:00-10:30 | f        |              | f
 测试老会员 | 2025-03-02    | evening    | 17:00-18:30 | t        |              | f
(4 rows)


会员额外签到次数:
       name       | extra_check_ins 
------------------+-----------------
 测试会员D        |               1
 测试新会员       |               3
 测试老会员       |               2
 测试额外签到会员 |               0
(4 rows)


会员卡状态:
       name       | card_type | card_category |  card_subtype  | remaining_group_sessions | remaining_private_sessions | valid_until 
------------------+-----------+---------------+----------------+--------------------------+----------------------------+-------------
 测试老会员       | group     | session       | 10_sessions    |                       10 |                            | 
 测试老会员       | group     | monthly       | single_monthly |                          |                            | 2025-04-01
 测试老会员       | private   | session       | 10_sessions    |                          |                         10 | 
 测试额外签到会员 | group     | session       | 10_sessions    |                        5 |                            | 
 测试额外签到会员 | private   | session       | 10_sessions    |                          |                          0 | 
(5 rows)


最近的错误日志:
psql:test_sql/run_all_tests.sql:83: ERROR:  column "created_at" does not exist
LINE 1: SELECT created_at, message
               ^

测试完成!
